### PR TITLE
TOC and portal page changes for clarity

### PR DESCRIPTION
**Pull request by rpetrusha**
# TOC and portal page changes for clarity

## Summary

Added TOC items for the main .NET Framework developer directory, removed MSDN links from the developer essentials page, added config file top-level top to app config TOC summary.

## Suggested Reviewers

@GuardRex @tompratt-AQ

### CODE REVIEW

File: docs/framework/development-guide.md [hunk 1]
```diff
@@ -66,14 +69,11 @@ This section explains how to create, configure, debug, secure, and deploy your .
...
-  
- [Advanced Reading for the .NET Framework](http://msdn.microsoft.com/en-us/faae8083-fecb-4514-b133-b0a5a32a7c3c)  
- Provides information about advanced development tasks and techniques in the .NET Framework, including extensibility, interoperability, and reflection. Also includes the reference topics for unmanaged APIs that can be used by managed apps, such as runtime hosts, compilers, disassemblers, debuggers, and profilers.  
+ <!--zz [Advanced Reading for the .NET Framework](http://msdn.microsoft.com/en-us/faae8083-fecb-4514-b133-b0a5a32a7c3c)  
+ Provides information about advanced development tasks and techniques in the .NET Framework, including extensibility, interoperability, and reflection. Also includes the reference topics for unmanaged APIs that can be used by managed apps, such as runtime hosts, compilers, disassemblers, debuggers, and profilers.  --> 
   
 ## Reference  
- [.NET Framework Class Library](http://go.microsoft.com/fwlink/p/?LinkID=227195)  
+ [.NET Framework Class Library](https://docs.microsoft.com/en-us/dotnet/api/?view=netframework-4.7)
```
Comment (GuardRex) [inline docs/framework/development-guide.md]: * Remove `en-us`
* Title:
*.NET Framework API Reference*
or
*.NET API Browser*

File: docs/framework/TOC.md [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
-# [Developing Client Applications with the .NET Framework](develop-client-apps.md)
-# [64-bit Applications](64-bit-apps.md)
+## [Application essentials](../standard/application-essentials.md)
+## [Data and modeling](data/index.md)
+## [Client applications](develop-client-apps.md)
+## [Web applications with ASP.NET](develop-web-apps-with-aspnet.md)
+## [Service-oriented applications with WCF](windows-services/index.md)
+## [Building workflows with Windows Workflow Foundation](windows-workflow-foundation/index.md)
+## [Parallel processing and concurrency](../standard/parallel-processing-and-concurrency.md)
+## [Network programming in the .NET Framework](network-programming/index.md)
```
Comment (GuardRex) [inline docs/framework/TOC.md]: This one uses the doc title, but most of these (e.g., *Application essentials*, *Data and modeling*, *Client applications*, etc.) are abbreviated titles. Make this *Network programming* to match?

File: docs/framework/TOC.md [hunk 3]
```diff
@@ -1,4 +1,18 @@
...
+## [64-bit Applications](64-bit-apps.md)
+## [Configuring .NET Framework apps](configure-apps/index.md)
```
Comment (GuardRex) [inline docs/framework/TOC.md]: Doc title: *Configuring Apps by using Configuration Files*

Maybe *Configuring apps* to match?

File: docs/framework/development-guide.md [hunk 4]
```diff
@@ -45,7 +48,7 @@ This section explains how to create, configure, debug, secure, and deploy your .
  [Network Programming in the .NET Framework](../../docs/framework/network-programming/index.md)  
  Describes the layered, extensible, and managed implementation of Internet services that you can quickly and easily integrate into your apps.  
   
- [Configuring .NET Framework Apps](http://msdn.microsoft.com/en-us/d789b592-fcb5-4e3d-8ac9-e0299adaaa42)  
+ [Configuring .NET Framework Apps](configure-apps/index.md)
```
Comment (GuardRex) [inline docs/framework/development-guide.md]: Doc title: *Configuring Apps by using Configuration Files*

**Review by guardrex — APPROVED** [review]

**Pull request merged**